### PR TITLE
Add map

### LINKED ISSUE #16: Add a map and show database on map
## Value proposition

**As a** user,
**I want to** have a map 
**so that** i can see where the different activities are located.

## Acceptance criteria

- [x] map should display the activities from database with marker on the map
- [x] marker should display onclick tooltip with name and location of the activity
- [x] Map will be accessible through basic navigation

## Complexity

medium

## Tasks

- [x] create a branch
- [x] import database to map
- [x] import location and name to the map
- [x] add basic navigation to map
- [x] add map to app

### PR DESCRIPTION
**Pull request by bensle**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -62,33 +63,37 @@ export default function App() {
 
   return (
     <Appcontainer>
-      {isHidden || (
+      {currentPage === 'activity' && (
```
Comment (macfrei) [inline src/App.js:66]: You have three different components which you render when `currentPage === 'activity'`. Why not extract these to an extra `ActivityPage.js` file and move them there? That way your App.js would be a bit leaner and cleaner :)

File: src/App.js [hunk 2]
```diff
@@ -62,33 +63,37 @@ export default function App() {
...
+        <TypeFilter onFilterTypeReset={filterTypeReset} onFilterTypeValue={setTypeFiltervalue} />
+      )}
+      {currentPage === 'activity' && (
         <ActivityCards
           activities={filteredActivities}
           onSetActivities={setActivities}
-          onSetIsHidden={() => setIsHidden(!isHidden)}
+          onSetIsHidden={() => setCurrentPage('favorites')}
+          onSetIsHidden1={() => setCurrentPage('map')}
```
Comment (macfrei) [inline src/App.js]: `onSetIsHidden` is not a good function name for what this is. Also, why do you also have `onSetIsHidden1`? I would suggest to have a function `onNavigate` which receives a parameter `page` as a string.
```suggestion
          onNavigate={(page) => setCurrentPage(page)}
```
In your `ActivityCard` you would then pass the parameter when clicking on a button:
```jsx
        <button onClick={() => onNavigate('favorites')}>Favorites</button>
        <button onClick={() => onNavigate('map')}>Map</button>
```
I know you wrote that you will do the navigation and stuff in the next user story, however, I would recommend you to change it.

File: src/App.js [hunk 3]
```diff
@@ -62,33 +63,37 @@ export default function App() {
...
           onDeleteActivity={deleteActivity}
           showModalConfirmation={showModalConfirmation}
         />
       )}
-      {isHidden && (
+      {currentPage === 'favorites' && (
         <FavoritesPage
           activities={activities}
-          onSetIsHidden={() => setIsHidden(!isHidden)}
+          onSetIsHidden={() => setCurrentPage('activity')}
```
Comment (macfrei) [inline src/App.js]: Same here with `onSetIsHidden` :)

File: src/pages/Map.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        {db.map(({id, name, infos, location, longitude, latitude}) => (
+          <Marker key={id} position={[latitude, longitude]}>
+            <Popup>
+              <Popupspan>{name}</Popupspan>
+              <br />
+              {location}
+              <br />
+              <br />
+              <a href={infos}>Find out more</a>
+              <br />
```
Comment (macfrei) [inline src/pages/Map.js]: I would recommend to use CSS for styling and not use more HTML elements for this. 
```suggestion
              {location}
              <a href={infos}>Find out more</a>
```

**Review by macfrei — COMMENTED** [review]
Good job, added a few comments :)

File: src/App.js (hunk 1 above)
Comment (bensle) [reply src/App.js:66]: Actually i wanted to do that within the next story. Navigation and routing

**Review by macfrei — APPROVED** [review]
Nice 🥳

**Pull request merged**